test/python/rest/submit_tests.py

```diff
@@ -0,0 +1,206 @@
+import os
+import sys
+import time
+import unittest
+import logging
+import streamsx.rest
+import streamsx.topology.context
+
+logger = logging.getLogger('submit_tests')
+logger.setLevel(logging.INFO)
+
+
+applicationName = 'TestApplication::Main'
+
+applicationBundle = './output/TestApplication.Main.sab'
+
+applicationParameters = {
+    'iterationInterval': 1.0,
+    'stringParameter': 'Hello, world',
+    'integerParameter': 42,
+    'floatParameter': 3.14159,
+    'booleanParameter': 'true',
+    'listParameter': [0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144],
+    'mapParameter': { 'one': 1, 'two': 2, 'three': 3, 'many': sys.maxsize }
+    }
+
+runInterval = 30
+
+def environment_variables_set():
+
+    variables = [ 'STREAMS_INSTALL', 'STREAMING_ANALYTICS_SERVICE_NAME', 'VCAP_SERVICES' ]
+    result = True
+
+    for variable in variables:
+        if variable not in os.environ:
+            logger.error(variable + ' environment variable not set')
+            result = False
+
+    return result
+
+@unittest.skipIf(not environment_variables_set() , "Streams environment variables not set")
+class RestSubmitTests(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(self):
+
+        # temporarily compile test application locally until REST API wrapper supports remote build service 
+        logger.warning('compile ' + applicationName)
+        result = os.system('sc -M ' + applicationName)
+        if not result==0: raise unittest.SkipTest(applicationName + ' failed to compile, exit code ' + str(result>>8) + ', signal ' + str(result&0xFF))
+
+        if not os.path.isdir('logs'):
+            os.mkdir('logs')
+
+        name = os.environ['STREAMING_ANALYTICS_SERVICE_NAME']
+
+        logger.warning('connect to IBM Cloud')
+        self.connection = streamsx.rest.StreamingAnalyticsConnection()
+
+        logger.warning('connect to service ' + name)
+        self.service = self.connection.get_streaming_analytics()
+
+        logger.warning('start service ' + name)
+        result = self.service.start_instance()
+        if not result['state']=='STARTED': raise unittest.SkipTest(name + ' service did not start')
+        if not result['status']=='running': raise unittest.SkipTest(name + ' service is not running')
+
+        logger.warning('check service ' + name)
+        instances = self.connection.get_instances()
+        if not len(instances)==1: raise unittest.SkipTest(name + ' service not found')
+        self.instance = instances[0]
+        if not self.instance.status=='running': raise unittest.SkipTest(name + ' service is not running')
+        if not self.instance.health=='healthy': raise unittest.SkipTest(name + ' service is not healthy')
+
+
+    def test_A_submit_with_defaults(self):
```

Comment:
Could you delete the `output` directory and the downloaded logs after the tests run? Other tests make sure that artifacts are deleted.

Reply:
Sure, no problem.